IdentityCore/src/requests/MSIDNonceTokenRequest.m

```diff
@@ -0,0 +1,205 @@
+//
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.  
+
+#import "MSIDNonceTokenRequest.h"
+#import "MSIDRequestParameters.h"
+#import "MSIDAuthority.h"
+#import "MSIDOpenIdProviderMetadata.h"
+#import "MSIDAccountIdentifier.h"
+#import "MSIDHttpRequest.h"
+#import "MSIDAADRequestConfigurator.h"
+
+@implementation MSIDNonceTokenRequest
+
+- (nullable instancetype)initWithRequestParameters:(nonnull MSIDRequestParameters *)parameters
+{
+    self = [super init];
+    if (self)
+    {
+        _requestParameters = parameters;
+    }
+    return self;
+}
+
+- (void)executeRequestWithCompletion:(nonnull MSIDNonceRequestCompletion)completionBlock
+{
+    MSIDCachedNonce *cachedNonce = [self.class getCachedNonceForKey:_requestParameters.authority.environment];
+    if (cachedNonce)
+    {
+        completionBlock(cachedNonce.nonce, nil);
+        return;
+    }
+
+    if (_requestParameters.authority.metadata.tokenEndpoint)
+    {
+        [self executeNetworkRequestWithCompletion:completionBlock];
+        return;
+    }
+    
+    [_requestParameters.authority resolveAndValidate:YES
+                                   userPrincipalName:_requestParameters.accountIdentifier.displayableId
+                                             context:_requestParameters
+                                     completionBlock:^(NSURL __unused *openIdConfigurationEndpoint, BOOL __unused validated, NSError *error)
+    {
+        if (error)
+        {
+            completionBlock(nil, error);
+            return;
+        }
+        
+        [self->_requestParameters.authority loadOpenIdMetadataWithContext:self->_requestParameters
```

Comment:
Is there a reason to use`->` instead of `.`

Reply:
No reason, code was copied from broker which was using ->. Updated to .